### ISSUE TITLE
React 18 dependency

### ISSUE THREAD

**Issue by tleo19** [mohitgupta8888/react-currency-format#36]
Hi, 

does react-currency-format plan on supporting react 18? 

<img width="899" alt="CleanShot 2022-08-06 at 13 49 56@2x" src="https://user-images.githubusercontent.com/65966422/183262232-9a6023c2-9193-4427-bb49-a7d108f39eba.png">

**Comment by hijal** [comment]
To fix this issue, you can ran `npm i react-currency-format --legacy-peer-deps`

**Comment by loganbutler** [comment]
> To fix this issue, you can ran npm i react-currency-format --legacy-peer-deps

Or they could just update the dependency. Takes 2 seconds.

**Comment by tleo19** [comment]
> > To fix this issue, you can ran npm i react-currency-format --legacy-peer-deps
> 
> Or they could just update the dependency. Takes 2 seconds.

right... this does not "fix" the issue it's more like bypassing the "error". 

Ended up removing `react-currency-format` and using `currency.js`

**Comment by loganbutler** [comment]
> Ended up removing react-currency-format and using currency.js

Same! Honestly after using currency.js feels like react-currency-format just ripped that and adapted it to a component.

**Comment by Sml995** [comment]
Just opened a PR, if this can help ...

**Comment by tomholford** [comment]
> Just opened a PR, if this can help ...

works for me, ty! 

use this in `package.json` until the maintainer updates the npm version:

```json
    "react-currency-format": "github:mohitgupta8888/react-currency-format#4f365ec5005e23451a3b95da1a209b0025622b8d",
```

**Comment by Sml995** [comment]
Thanks !

**Comment by gysenlionel** [comment]
You can use if you force 

npm install react-currency-format --save --force

or you can also use

npm i react-currency-format --save --force

**Comment by surfjedi** [comment]
+1

**Comment by ciobiano** [comment]
> > Ended up removing react-currency-format and using currency.js
> 
> Same! Honestly after using currency.js feels like react-currency-format just ripped that and adapted it to a component.

Please how could i apply this to my project? any hint?

**Comment by loganbutler** [comment]
> Please how could i apply this to my project? any hint?

If you just want to use currency.js, you can simply go to this link and follow the instructions. https://currency.js.org/
Otherwise just follow the instructions above for using the PR made to fix the peer dependency issue. I personally opted to just not use it for now.

for vanilla currency.js; 
`import currency from 'currency.js'`
then
`const price = currency(10).add(10)` or whatever you want.

Also...I noticed the maintainer approved the PR but it's still not updated? Amazing! Thanks to @Sml995 for taking his time to make that PR. I would like to use this but don't like ugly peer deps in my terminal.

**Comment by h0jeZvgoxFepBQ2C** [comment]
Please merge this PR and release a new version!!! Thank you!

**Comment by ricardogomez-dev** [comment]
Waiting for the pull request...

**Comment by lakshmiprasad-apisod** [comment]
Waiting for newer version
